src/test/java/com/jcabi/beanstalk/maven/plugin/GenericMojoTest.java

```diff
@@ -0,0 +1,115 @@
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.jcabi.beanstalk.maven.plugin;
+
+import java.io.File;
+import java.util.zip.ZipFile;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.yaml.snakeyaml.error.YAMLException;
+
+/**
+ * Test case for {@link AbstractBeanstalkMojo}.
+ * @author Neo Matrix (Neo.matrix@gmail.com)
+ * @version $Id$
+ */
+public final class GenericMojoTest {
+    /**
+     * {@link AbstractBeanstalkMojo} can execute successfully.
+     * @throws Exception If something is wrong.
+     * @checkstyle ExecutableStatementCountCheck (40 lines)
+     */
+    @Test
+    public void executesSuccessfully() throws Exception {
+        // @checkstyle IllegalTypeCheck (2 lines)
+        final AbstractBeanstalkMojo mojo =
+            Mockito.mock(AbstractBeanstalkMojo.class);
+        final File mockFile = Mockito.mock(File.class);
+        final ZipFile mockZipFile = Mockito.mock(ZipFile.class);
+        Mockito.when(mockFile.exists()).thenReturn(true);
+        Mockito.when(mojo.createZipFile()).thenReturn(mockZipFile);
+        Mockito.doCallRealMethod().when(mojo).execute();
+        Mockito.doCallRealMethod().when(mojo).getLog();
+        Mockito.doCallRealMethod().when(mojo)
+            .setWar(Mockito.any(File.class));
+        Mockito.doCallRealMethod().when(mojo)
+            .setName(Mockito.any(String.class));
+        Mockito.doCallRealMethod().when(mojo)
+            .setBucket(Mockito.any(String.class));
+        Mockito.doCallRealMethod().when(mojo)
+            .setKey(Mockito.any(String.class));
+        mojo.setWar(mockFile);
+        mojo.setName("name");
+        mojo.setBucket("bucket");
+        mojo.setKey("key");
+        mojo.execute();
+        Mockito.verify(mojo).createZipFile();
+        Mockito.verify(mojo).validate(mockZipFile);
+        Mockito.verify(mojo).createWarFile(mockZipFile);
+        Mockito.verify(mojo).validateWarFile(Mockito.any(WarFile.class));
+        Mockito.verify(mojo).createServerCredentials();
+        Mockito.verify(mockZipFile).close();
+        Mockito.verify(mockFile, Mockito.times(2)).exists();
+    }
+
+    /**
+     * {@link AbstractBeanstalkMojo} can validate bad yaml.
+     * @throws Exception If something is wrong
+     */
+    @Test(expected = YAMLException.class)
+    public void validatesBadYaml() throws Exception {
+        final String invalid = new StringBuilder()
```

Comment:
@tmjee this variable is redundant, see http://www.yegor256.com/2015/09/01/redundant-variables-are-evil.html